### PR TITLE
Radio Module B update and xdot/sarau260 userland driver code

### PR DESCRIPTION
**Pull request by adkinsjd**
This pull request includes the code necessary for minimal operation of the xdot and the u260 along with two minimal tests of their operation. It also includes an updated board file (the old board revision was moved to the old_revisions folder).

### CODE REVIEW

File: software/apps/libsignpost/sara_u260.c [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        //nothign to do about this - return an error
+        return SARA_U260_ERROR;
+    }
+
+    return SARA_U260_SUCCESS;
+}
+
+static int sara_u260_del_file(const char* fname) {
+    char c[50];
+    int clen = sprintf(c, "AT+UDELFILE=\"%s\"\r", fname);
```
Comment (bradjc) [inline software/apps/libsignpost/sara_u260.c]: These should be "n" functions, if for no other reason than to set a good example.

File: software/apps/libsignpost/sara_u260.c [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        return SARA_U260_ERROR;
+    }
+
+    //now manually memcpy out the data into tbuf (because there could be nulls)
+    memcpy(buf,tbuf+(len - 7 - dlen),dlen);
+
+    return dlen;
+}
+
+
```
Comment (bradjc) [inline software/apps/libsignpost/sara_u260.c]: Delete extra whitespace

File: software/apps/libsignpost/xdot.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+//dr >= 0 and dr<=4
+//-1 on error
+int xdot_get_txdr(void);
+
+//0<=txpwr<=20
+int xdot_set_txpwr(uint8_t tx);
+
+//0 or 1
+int xdot_set_adr(uint8_t adr);
```
Comment (bradjc) [inline software/apps/libsignpost/xdot.h]: What does this function do?

File: software/apps/libsignpost/xdot.h [hunk 4]
```diff
@@ -0,0 +1,38 @@
+#pragma once
+
```
Comment (bradjc) [inline software/apps/libsignpost/xdot.h:4]: A one liner comment at the top of the .h files with the name of the chip and a URL would be great.

File: software/apps/radio_module/cellular_test/main.c [hunk 5]
```diff
@@ -0,0 +1,47 @@
+#include <string.h>
```
Comment (bradjc) [inline software/apps/radio_module/cellular_test/main.c:1]: A couple two line READMEs that explains what should happen in these tests would be great.

File: software/kernel/signpost_drivers/src/sara_u260.rs [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+                    cb.schedule(max_idx, (buf.as_ptr() as usize), 0);
+                });
+
+                rb
+            });
+        });
+
+
+
+        // old readline code
```
Comment (bradjc) [inline software/kernel/signpost_drivers/src/sara_u260.rs]: Don't need this old code.

File: software/kernel/signpost_drivers/src/xdot.rs [hunk 7]
```diff
@@ -0,0 +1,402 @@
...
+                    cb.schedule(max_idx, (buf.as_ptr() as usize), 0);
+                });
+
+                rb
+            });
+        });
+
+
+
+        // old readline code
```
Comment (bradjc) [inline software/kernel/signpost_drivers/src/xdot.rs]: here too

**Comment by adkinsjd** [comment]
These commits should address all of Brad's comments.

File: software/apps/libsignpost/multi_console.c [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+void   console_read_async(int console_num, uint8_t* buf, size_t max_len, subscribe_cb cb) {
+    allow(console_num, 0, (void*)buf, max_len);
+    subscribe(console_num, 2, cb, NULL);
+}
+
+size_t console_write(int console_num, uint8_t* buf, size_t count) {
+
+    uint8_t* cbuf = (uint8_t*)malloc(count * sizeof(uint8_t));
```
Comment (brghena) [inline software/apps/libsignpost/multi_console.c:44]: Here's the malloc, but where's the free? (same goes for the other three malloc uses)

File: software/kernel/signpost_drivers/src/xdot.rs [hunk 9]
```diff
@@ -0,0 +1,327 @@
...
+                    // only both receiving if we've got somewhere to put it
+                    app.read_buffer = app.read_buffer.take().map(|app_buf| {
+                        self.rx_buffer.take().map(|buffer| {
+                            /*
+                            let mut receive_len = app_buf.len();
+                            if receive_len > buffer.len() {
+                                receive_len = buffer.len();
+                            }*/
+                            //XXX: this could receive more than the app buffer...
+                            self.uart.receive_automatic(buffer, 10);
```
Comment (brghena) [inline software/kernel/signpost_drivers/src/xdot.rs:200]: This and the `receive_automatic` call in sara_u260 have different timeouts. You should put a comment to note the difference and explain why it changed from the console driver (in whichever changed).

**Review by brghena — CHANGES_REQUESTED** [review]
Overall looks really good. I've got two changes, but after those I'd say it's good to merge.

**Comment by adkinsjd** [comment]
@brghena Your changes are in the code

**Comment by brghena** [comment]
I'd say it's good to go then when it passes travis.

**Pull request merged**